### PR TITLE
chore: Prepare for multiplatform build

### PR DESCRIPTION
**Pull request by nahsi**

### CODE REVIEW

File: .github/workflows/update_fluence.yml [hunk 1]
```diff
@@ -1,37 +1,24 @@
...
       - name: Update Fluence in fluence.json to ${{ github.event.inputs.version }}
-        run: |
-          cat <<FLUENCE > fluence/fluence.json
-          {
-            "version": "${{ github.event.inputs.version }}",
-            "url": "${{ github.event.inputs.url }}",
-            "sha256": "${{ github.event.inputs.sha256 }}"
-          }
-          FLUENCE
+        run: echo ${{ inputs.version }} | jq > fluence/fluence.json
```
Comment (folex) [inline .github/workflows/update_fluence.yml:21]: what about sha check an all that?

File: Dockerfile [hunk 2]
```diff
@@ -111,7 +108,8 @@ ENV FLUENCE_ENV_AQUA_IPFS_EXTERNAL_API_MULTIADDR=/ip4/127.0.0.1/tcp/5001
 ENV FLUENCE_ENV_AQUA_IPFS_LOCAL_API_MULTIADDR=/ip4/127.0.0.1/tcp/5001
 
 # download fs-repo-migrations
-RUN wget -qO - "https://dist.ipfs.io/fs-repo-migrations/v2.0.2/fs-repo-migrations_v2.0.2_linux-amd64.tar.gz" | tar -C /usr/local/bin --strip-components=1 -zxvf -
+COPY docker/download_fs_repo_migrations.sh /docker/download_fs_repo_migrations.sh
```
Comment (folex) [inline Dockerfile:111]: let's name the script differently: fs_migrations => ipfs_migrations

File: .github/workflows/update_fluence.yml (hunk 1 above)
Comment (nahsi) [reply .github/workflows/update_fluence.yml:21]: What sha check? I don't understand.

File: fluence/download_fluence.sh [hunk 3]
```diff
@@ -1,24 +1,62 @@
...
+        ;;
+    *)
+        echo "Unsupported architecture: $ARCHITECTURE. Exiting..."
+        exit 1
+        ;;
+esac
+
+# Fetch URL, SHA256, and version based on detected architecture
+URL=$(jq -r ".${KEY}.url" $CONFIG)
+SHA256=$(jq -r ".${KEY}.sha256" $CONFIG)
```
Comment (folex) [inline fluence/download_fluence.sh:38]: ah, it is now here, good

File: s6/minimal/run_fluence [hunk 4]
```diff
@@ -5,7 +5,7 @@
 with-contenv
 
 # remove old HOME variable from the environment, so the daemon isn't confused by it
-# see https://github.com/fluencelabs/node-distro/issues/14 for more details
+# see https://github.com/fluencelabs/rust-peer-distro/issues/14 for more details
```
Comment (folex) [inline s6/minimal/run_fluence:8]: nox distro

File: s6/minimal/run_fluence [hunk 5]
```diff
@@ -5,7 +5,7 @@
...
 unexport HOME
-# 'setuidgid abc' runs '/usr/bin/fluence' as user 'abc'
-s6-setuidgid abc /usr/bin/fluence $@
+# 'setuidgid abc' runs '/usr/bin/rust-peer' as user 'abc'
```
Comment (folex) [inline s6/minimal/run_fluence:10]: nox

File: .github/workflows/update_fluence.yml [hunk 6]
```diff
@@ -1,37 +1,24 @@
-name: "update_fluence"
+name: "Update rust-peer version"
```
[the remark this answers is not in the document]
Comment (nahsi) [reply .github/workflows/update_fluence.yml:1]: Well okay, but only here?
```> grep rust-peer -R | wc -l
grep: .git/index: binary file matches
168
```

**Review by folex — APPROVED** [review]
very nice!

File: .github/workflows/update_fluence.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (folex) [reply .github/workflows/update_fluence.yml:1]: wherever you see it, but not in the git history :)

**Pull request merged**